### PR TITLE
[KOGITO-983] Check first that the CLI exists

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-983

Change has been done as well how the smoke tests are started. No more direct call to godog application but call of go test. That means that we don't need to install godog binary anymore. 
Also, it gives us much more flexibility of configuration (--smoke.* options) and setup.

Also added `timeout` and `debug` options.

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

File: test/smoke/framework/cli.go [hunk 1]
```diff
@@ -16,9 +16,23 @@ package framework
...
 
+// CheckCliBinaryExist checks if the CLI binary does exist
+func CheckCliBinaryExist() (bool, error) {
+	path, err := getEnvOperatorCliPath()
+	if err != nil {
+		return false, err
+	}
+
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		return false, nil
```
Comment (sutaakar) [inline test/smoke/framework/cli.go]: Please cover also branch when err is not nil but also not IsNotExist()

Comment (radtriste) [reply test/smoke/framework/cli.go]: updated

File: test/smoke/main_test.go [hunk 2]
```diff
@@ -15,12 +15,64 @@
...
+
+	if !strings.Contains(opt.Tags, defaultDisabledTag) {
+		if opt.Tags != "" {
+			opt.Tags += " && "
+		}
+		opt.Tags += defaultDisabledTag
+	}
+
+	// Check CLI binary is existing
+	if exits, err := framework.CheckCliBinaryExist(); err != nil {
```
Comment (sutaakar) [inline test/smoke/main_test.go]: This means that CLI binary is required for running any test, right?

Comment (radtriste) [reply test/smoke/main_test.go]: yes... will modify that...

File: .circleci/config.yml [hunk 3]
```diff
@@ -15,3 +15,6 @@ jobs:
       - run: go vet ./...
       - run: CGO_ENABLED=0 go test ./cmd/... -count=1 -p=1
       - run: CGO_ENABLED=0 go test ./pkg/... -count=1 -p=1
+      - run: git clone https://github.com/bats-core/bats-core.git
```
Comment (radtriste) [inline .circleci/config.yml]: @spolti It seems that the circle ci job is only running the `build` job (https://circleci.com/gh/kiegroup/kogito-cloud-operator/783#build-parameters/containers/0)
So for now, I included it into the `build` job, but I would like better to have a second job, like `bats-tests` job.
Do you know how we can update that ?

File: test/smoke/main_test.go [hunk 4]
```diff
@@ -100,3 +112,12 @@ func deleteNamespaceIfExists(namespace string) {
 		}
 	}
 }
+
+func matchingFeature(tags string, features []*gherkin.Feature) bool {
```
Comment (sutaakar) [inline test/smoke/main_test.go]: tags parameter is unused

File: .circleci/config.yml (hunk 3 above)
Comment (spolti) [reply .circleci/config.yml]: @radtriste for use multiple jobs, then a workflow is necessary:

https://circleci.com/docs/2.0/workflows/

but, in this case, I think that a sequential steps is ok, or, is there a specific reason to use multiple jobs for it?

**Comment by spolti** [comment]
@radtriste is it ready to merge or are you going to modify the circleci job?

File: .circleci/config.yml (hunk 3 above)
Comment (radtriste) [reply .circleci/config.yml]: If you are ok with the current solution, we can stick to it (and refactor later if needed)

**Comment by radtriste** [comment]
@sutaakar are you ok with that ?

File: .circleci/config.yml (hunk 3 above)
Comment (spolti) [reply .circleci/config.yml]: It is ok for me

**Comment by radtriste** [comment]
Rebased and squashed

**Pull request merged**